File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.java
import java.util.*;

class Solution {
    public String[] solution(String[] strings, int n) {
        
        for (int i = 0; i < strings.length; i++) {

            String s = strings[i];
            String sub = String.valueOf(s.charAt(n));
            strings[i] = sub + s;
        }

        Arrays.sort(strings);

        for (int i = 0; i < strings.length; i++) {
            
            strings[i] = strings[i].substring(1);
        }

        return strings;
    }
}

File: Kotlin/프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.kt
class Solution {
    fun solution(strings: Array<String>, n: Int): Array<String> {
        var answer = strings.copyOf()

        for (i in answer.indices) {

            answer[i] = answer[i][n] + answer[i]
        }

        answer.sort()

        for (i in answer.indices) {
            answer[i] = answer[i].substring(1)
        }

        return answer
    }
}